src/ping_maker.js

```diff
@@ -161,8 +161,8 @@ class PingMaker {
         // Attempt to fetch the addon version, if we're a
         // webextension.
         var browser =
-            (typeof browser !== "undefined") ? browser : (typeof chrome !== "undefined" ? chrome : null);
-        if (browser) {
+            (typeof browser !== "undefined") ? browser : (typeof chrome !== "undefined" ? chrome : {});
+        if (typeof browser.runtime !== "undefined" && typeof browser.runtime.getManifest !== undefined) {
```

Comment:
We should have a utility function to do this `typeof ... !== "undefined"` that is required for QML to work, but it is boring to write and ugly to read.